### PR TITLE
Tile reorganization, column selector fix

### PR DESCRIPTION
**Pull request by rpiaggio**
This PR started out as an attempt to fix https://app.shortcut.com/lucuma/story/3639/column-selector-drop-downs-for-summary-tables-don-t-update, which it does, but then I went into the rabbit hole of aligning the controls in the tile titles and from there to the even deeper rabbit hole of organizing tile code.

The result is that each tile now has all of its code in only one file, with `Body` and `Title` as private components, a `TileState` for the shared state among them, and an `apply` method that builds the tile and passes to each subcomponent only what they need from the state.

This eliminates the files that only contained glue code to build the tiles (that code is now the new `apply` methods).

Styling of titles has been simplified a bit too.

Oh, and now the column selector checkboxes show whether the column is visible :D

### CODE REVIEW

File: common/src/main/scala/explore/components/Tile.scala [hunk 1]
```diff
@@ -135,31 +135,22 @@ object Tile:
...
           )(
             // Tile title
-            <.div(ExploreStyles.TileTitle)(
-              ^.key := s"tileTitle-${p.id.value}",
+            <.div(ExploreStyles.TileTitle, ^.key := s"tileTitle-${p.id.value}")(
               // Title and optional back button
-              <.div(
-                ExploreStyles.TileTitleMenu |+| p.tileTitleClass,
+              <.div(ExploreStyles.TileTitleMenu |+| p.tileTitleClass)(
                 p.back.map(b => <.div(ExploreStyles.TileButton, b)),
```
Comment (cquiroz) [inline common/src/main/scala/explore/components/Tile.scala:141]: is it correct to move the back  button to the title text?

File: explore/src/main/scala/explore/config/ConfigurationTile.scala [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+        selectedConfig,
+        revertedInstrumentConfig,
+        modes,
+        sequenceChanged,
+        readonly,
+        units
+      )
+    )
+
+  private case class Body(
```
Comment (cquiroz) [inline explore/src/main/scala/explore/config/ConfigurationTile.scala:99]: Seems a good refactoring in general

File: explore/src/main/scala/explore/observationtree/ObsSummaryTile.scala [hunk 3]
```diff
@@ -74,87 +78,134 @@ object ObsSummaryTable:
...
-      PriorityColumnId      -> "Priority",
       RAColumnId            -> "RA",
       DecColumnId           -> "Dec",
-      TimingWindowsColumnId -> "Scheduling Windows",
       SEDColumnId           -> "SED",
-      ChargedTimeColumnId   -> "ChargedTime"
+      ConstraintsColumnId   -> "Constraints",
+      ConfigurationColumnId -> "Configuration",
+      DurationColumnId      -> "Duration"
+      // ValidationCheckColumnId -> " ",
```
Comment (cquiroz) [inline explore/src/main/scala/explore/observationtree/ObsSummaryTile.scala:146]: Commented code

**Review by cquiroz — APPROVED** [review]
In general I'm ok with the refactor. It is hard to tell about the change on the dropdown among the other changes

LGTM

File: explore/src/main/scala/explore/observationtree/ObsSummaryTile.scala (hunk 3 above)
Comment (rpiaggio) [reply explore/src/main/scala/explore/observationtree/ObsSummaryTile.scala:146]: We have commented the columns that are pending to add. I just consolidated them.

File: common/src/main/scala/explore/components/Tile.scala (hunk 1 above)
Comment (rpiaggio) [reply common/src/main/scala/explore/components/Tile.scala:141]: This didn't change. I just separated style and content into two different parameter lists (which I find it resembles HTML's structure of attributes together with the tag and then content in a different block).

**Pull request merged**